ui/apps/platform/src/global/initializeAnalytics.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable */
+
+// the code below is generated from segment api with the exception of the writeKey parameter
+// segment intends for the write key to be hardcoded but we are pulling it from the telemetry config service call and adding it using a parameter
+
+export function initializeSegment(writeKey) {
+    const analytics = (window.analytics = window.analytics || []);
```

Comment:
Related to the method call question below in analytics.ts file, is there a possible time interval between this assignment to `window.analytics` and addition of methods in line 45 below?

Reply:
Isn't this part happening synchronously though? So outside of this code the methods would either exist on the object or the object would be undefined. We're calling `window.analytics.identify` immediately after calling `initializeSegment` in `telemetryConfig.js` without issue

Reply:
Yes, thank you for thinking it through!